RELEASE/scripts/autoscend/iotms/mr2022.ash

```diff
@@ -402,4 +419,140 @@ boolean auto_handleParka()
 	equip($item[jurassic parka]); // already configured, just equip
 
 	return get_property("parkaMode") == tempDino && have_equipped($item[jurassic parka]);
+}
+
+boolean auto_hasAutumnaton()
+{
+	return get_property("hasAutumnaton").to_boolean();
+}
+
+// only valid when autumnaton is not current out on a quest
+boolean auto_autumnatonCanAdv(location canAdventureInloc)
+{
+	if(!auto_hasAutumnaton())
+	{
+		return false;
+	}
+
+	if(canAdventureInloc == $location[8-bit realm] && possessEquipment($item[continuum transfunctioner]) && auto_is_valid($item[continuum transfunctioner]))
+	{
+		equip($item[continuum transfunctioner]);
+	}
+
+	foreach index,loc in get_autumnaton_locations()
+	{
+		if(loc == canAdventureInloc)
+		{
+			return true;
+		}
+	}
+	return false;
+}
+
+boolean auto_autumnatonReadyToQuest()
+{
+	if(!auto_hasAutumnaton())
+	{
+		return false;
+	}
+
+	return item_amount($item[autumn-aton]) != 0;
+}
+
+location auto_autumnatonQuestingIn()
+{
+	return to_location(get_property("autumnatonQuestLocation"));
+}
+
+boolean auto_autumnatonCheckForUpgrade(string upgrade)
+{
+	string currentUpgrades = get_property("autumnatonUpgrades");
+	if(contains_text(currentUpgrades,upgrade))
+	{
+		return true;
+	}
+	return false;
+}
+
+boolean auto_sendAutumnaton(location loc)
+{
+	if(auto_autumnatonCanAdv(loc))
+	{
+		cli_execute("autumnaton send " + loc);
+		handleTracker("Autumnaton sent to " + loc, "auto_otherstuff");
+		return true;
+	}
+	return false;
+}
+
+void auto_autumnatonQuest()
+{
+	if(!auto_autumnatonReadyToQuest()) return;
+
+	// complete any pending upgrades if it just returned
+	if (total_turns_played() == get_property("autumnatonQuestTurn").to_int() + 1)
+	{
+		catch cli_execute("autumnaton upgrade");
+	}
+
+	// prioritize getting important upgrades
+	if(!auto_autumnatonCheckForUpgrade("leftarm1"))
+	{
+		if(auto_sendAutumnaton($location[The Haunted Pantry]))
+		{
+			return;
+		}
+		else
+		{
+			abort("Haunted pantry should always be available for autumnaton, but autoscend determined it is not. Report issue.");
+		}
+	}
+
+	if(!auto_autumnatonCheckForUpgrade("leftleg1"))
+	{
+		// some bat zones may not be adventured in, so try them all
+		if(auto_sendAutumnaton($location[Guano Junction])) return;
+		if(auto_sendAutumnaton($location[The Batrat And Ratbat Burrow])) return;
+		if(auto_sendAutumnaton($location[The Beanbat Chamber])) return;
+	}
+
+	if(!auto_autumnatonCheckForUpgrade("rightleg1"))
+	{
+		if(auto_sendAutumnaton($location[The Haunted Library])) return;
+	}
+
+	// acquire items to help quests
+	if(fastenerCount() < 30 && lumberCount() < 30)
+	{
+		if(auto_sendAutumnaton($location[The Smut Orc Logging Camp])) return;
```

Comment:
possibly worth adding in a "if do not have right arm" function that sends to overgrown lot as a back up, in case someone hasnt levelled high enough yet?